implementation/src/main/java/io/smallrye/mutiny/operators/UniCache.java

```diff
@@ -18,12 +19,19 @@
     private static final int COMPLETED = 3;
     private final AtomicReference<UniSubscription> subscription = new AtomicReference<>();
     private final List<UniSubscriber<? super I>> subscribers = new ArrayList<>();
+    private final BiPredicate<I, Throwable> validator;
     private int state = 0;
     private I item;
     private Throwable failure;
 
     UniCache(Uni<? extends I> upstream) {
```

Comment:
Would you prefer to see the default predicate in the `AbstractUni` and go only with the new constructor?